ibm/resource_ibm_is_virtual_endpoint_gateway.go

```diff
@@ -0,0 +1,333 @@
+package ibm
+
+import (
+	"log"
+	"time"
+
+	"github.com/IBM/go-sdk-core/core"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	isVirtualEndpointGatewayName               = "name"
+	isVirtualEndpointGatewayResourceType       = "resource_type"
+	isVirtualEndpointGatewayResourceGroupID    = "resource_group"
+	isVirtualEndpointGatewayCreatedAt          = "created_at"
+	isVirtualEndpointGatewayIPs                = "ips"
+	isVirtualEndpointGatewayIPsID              = "id"
+	isVirtualEndpointGatewayIPsAddress         = "address"
+	isVirtualEndpointGatewayIPsName            = "name"
+	isVirtualEndpointGatewayIPsSubnet          = "subnet"
+	isVirtualEndpointGatewayIPsSubnetID        = "subnet_id"
+	isVirtualEndpointGatewayIPsResourceType    = "resource_type"
+	isVirtualEndpointGatewayHealthState        = "health_state"
+	isVirtualEndpointGatewayLifecycleState     = "lifecycle_state"
+	isVirtualEndpointGatewayTarget             = "target"
+	isVirtualEndpointGatewayTargetName         = "name"
+	isVirtualEndpointGatewayTargetResourceType = "resource_type"
+	isVirtualEndpointGatewayVpcID              = "vpc"
+)
+
+func resourceIBMISEndpointGateway() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMisVirtualEndpointGatewayCreate,
+		Read:     resourceIBMisVirtualEndpointGatewayRead,
+		Update:   resourceIBMisVirtualEndpointGatewayUpdate,
+		Delete:   resourceIBMisVirtualEndpointGatewayDelete,
+		Exists:   resourceIBMisVirtualEndpointGatewayExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			isVirtualEndpointGatewayName: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateISName,
+				Description:  "Endpoint gateway name",
+			},
+			isVirtualEndpointGatewayResourceType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Endpoint gateway resource type",
+			},
+			isVirtualEndpointGatewayResourceGroupID: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "The resource group id",
+			},
+			isVirtualEndpointGatewayCreatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Endpoint gateway created date and time",
+			},
+			isVirtualEndpointGatewayHealthState: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Endpoint gateway health state",
+			},
+			isVirtualEndpointGatewayLifecycleState: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Endpoint gateway lifecycle state",
+			},
+			isVirtualEndpointGatewayIPs: {
```

Comment:
From API docs isVirtualEndpointGatewayIPs is a required attribute

Reply:
I can see isVirtualEndpointGatewayIPs(ips) as an optional attribute.
https://pages.github.ibm.com/riaas/api-spec/spec_aspirational/#/Endpoint%20gateways/create_endpoint_gateway